data_filter_mongodb/README.md

```diff
@@ -0,0 +1,316 @@
+    "Email": "rob@opa.com",
+    "Mobile": "1343238738",
+    "Manager": "john "
+  },
+  {
+    "Name": "danerys",
+    "Designation": "director of engineering",
+    "Salary": 350000,
+    "Email": "danerys@opa.com",
+    "Mobile": "12332423738",
+    "Manager": "nil"
+  }
+]
+```
+
+### Querying opa-mongo server:
+
+Below is the request you can make to opa server:
+
+This requests for employee john details as we notice request param and path in request body says john, and the request body expects the requested user name. Based on the rego policy the requested user should be either the employee or the respective manager to retrieve the employee details.
+
+```
+URL: http://localhost:9095/employees/john
+```
+Request body:
+```
+{
+    "input": {
+        "method": "GET",
+        "path": ["employees", "john"],
+        "user": "danerys"
+    }
+}
+```
+
+Response:
+```
+{
+    "result": {
+        "Defined": true,
+        "Data": [
+            {
+                "Name": "john",
+                "Designation": "lead engineer",
+                "Salary": 270000,
+                "Email": "john@opa.com",
+                "Mobile": "1233743438738",
+                "Manager": "danerys"
+            }
+        ]
+    }
+}
+```
+
+In opa-mongo server logs you can observe the received query, equivalent opa-query and the equivalent mongo-query based on the rego policy:
+
+```
+2020/08/07 18:15:45 received requestmap[method:GET path:[employees john] user:danery]
+2020/08/07 18:15:45 opa-query: ["danery" = data.employees[_].name; "john" = data.employees[_].name "danery" = data.employees[_].manager; "john" = data.employees[_].name]
+2020/08/07 18:15:45 mongo query: map[$or:[map[$and:[map[name:map[$eq:danery]] map[name:map[$eq:john]]]] map[$and:[map[manager:map[$eq:danery]] map[name:map[$eq:john]]]]]]
+```
+
+Now you have successfully retrieved john details by requesting the opa-mongo server by enforcing the policies from the mongo database.
+
+### Supported OPA Built-in Functions:
+
+#### Comparison:
+
+- [x] ==
+- [x] !=
+- [x] <
+- [x] <=
+- [x] >
+- [x] >=
+
+## Support for OPA references
+
+References are used to access nested documents in OPA. OPA policies can be written over deeply nested structures which the server would then translate to Mongo `Nested` queries.
+
+## Generated Mongo queries
+
+For the OPA operators mentioned above, following are Mongo queries generated by the server:
+
+### Term level Queries
+
+- Term Query
+- Range Query
+
+### Joining Queries
+
+- Nested Query
+
+### Compound Queries
+
+- Bool Query
+
+### Limitations:
+
+- The OPA policies should be written according to the fields in the MongoDB documents to get the desired results.
+- The server supports limited OPA operators and returns an empty result if the OPA policy contains an unsupported operator and logs the error.
+- The server supports only two endpoints `/employees` and `/employees/{employee_name}` for fetching posts created when the server starts.
```

Comment:
The README looks good. Can you add a section about `How this (OPA-MongoDB Data filtering) works`. It would be good to document that.